api/observation.go

```diff
@@ -0,0 +1,410 @@
+package api
+
+import (
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/ONSdigital/dp-graph/v2/observation"
+	errs "github.com/ONSdigital/dp-observation-api/apierrors"
+	"github.com/ONSdigital/dp-observation-api/models"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+)
+
+// Upper limit, if this is not big enough, we may need to consider increasing value
+// and then if this has a performance hit then consider paging
+const (
+	defaultObservationLimit = 10000
+	defaultOffset           = 0
+
+	getObservationsAction = "getObservations"
+)
+
+var (
+	observationNotFound = map[error]bool{
+		errs.ErrDatasetNotFound:      true,
+		errs.ErrEditionNotFound:      true,
+		errs.ErrVersionNotFound:      true,
+		errs.ErrObservationsNotFound: true,
+	}
+
+	observationBadRequest = map[error]bool{
+		errs.ErrTooManyWildcards: true,
+	}
+)
+
+type observationQueryError struct {
+	message string
+}
+
+func (e observationQueryError) Error() string {
+	return e.message
+}
+
+func errorIncorrectQueryParameters(params []string) error {
+	return observationQueryError{
+		message: fmt.Sprintf("incorrect selection of query parameters: %v, these dimensions do not exist for this version of the dataset", params),
+	}
+}
+
+func errorMissingQueryParameters(params []string) error {
+	return observationQueryError{
+		message: fmt.Sprintf("missing query parameters for the following dimensions: %v", params),
+	}
+}
+
+func errorMultivaluedQueryParameters(params []string) error {
+	return observationQueryError{
+		message: fmt.Sprintf("multi-valued query parameters for the following dimensions: %v", params),
+	}
+}
+
```

Comment:
Consider moving this into a separate `errors.go` file in this package.

Reply:
I can move these functions to apierrors/errors.go